test/runtests.jl

```diff
@@ -1012,4 +1012,16 @@ end
         @test U == UpperTriangular(ones(3,3))
         @test axes(U) == (Base.OneTo(3),Base.OneTo(3))
     end
-end
\ No newline at end of file
+end
+
+@testset "Trues" begin
+    @test Trues(2,3) == Trues((2,3)) == trues(2,3)
+    @test Falses(2,3) == Falses((2,3)) == falses(2,3)
+    dim = (4,5)
+    mask = Trues(dim)
+    x = randn(dim)
+    @test x[mask] == vec(x) # getindex
+    y = similar(x)
+    y[mask] = x # setindex!
+    @test y == x
```

Comment:
Add `@test_throws BoundsError ...` to test case where axes don't match.

Reply:
done, only took 4 commits to get it right (i hope)...